### PR TITLE
use service instead of plist

### PR DESCRIPTION
**Pull request by glepnir**
recently homebrew had deprecated the `plist_options` suggest use the `service.require_root` instead . I am not familiar the homebrew just read the doc of it then write this.  not sure is correct so i only write the yabai .if this is no problem i can  add skhd  thanks.

### CODE REVIEW

**Comment by mvllow** [comment]
I've never installed a formulae locally so may be an issue on my end but I'm getting the following error when cloning this PR and running `brew install ./yabai.rb`:

```sh
Service#environment_variables expects a hash
Error: Failed to install service files
Error: Service#environment_variables expects a hash
```

File: yabai.rb [hunk 1]
```diff
@@ -39,41 +39,15 @@ def caveats; <<~EOS
...
-      <string>Interactive</string>
-      <key>Nice</key>
-      <integer>-20</integer>
-    </dict>
-    </plist>
-    EOS
+  service do
+    run opt_bin/"yabai"
+    require_root true
+    environment_variables HOMEBREW_PREFIX/"bin:/usr/bin:/bin:/usr/sbin:/sbin"
```
Comment (mvllow) [inline yabai.rb:45]: Saw this used in a few other formulae and it seems to fix the error.

```suggestion
    environment_variables PATH: "#{HOMEBREW_PREFIX}/bin:/usr/bin:/bin:/usr/sbin:/sbin"
```

File: yabai.rb [hunk 2]
```diff
@@ -39,41 +39,15 @@ def caveats; <<~EOS
...
+    keep_alive true
+    interval 30
+    log_path var/"log/yabai/yabai.out.log"
+    err_log_path var"/log/yabai/yabai.err.log"
```
Comment (mvllow) [inline yabai.rb:49]: Looks like this option is called `error_log_path`.
https://docs.brew.sh/Formula-Cookbook#page

```suggestion
    error_log_path "#{var}/log/yabai/yabai.err.log"
```

File: yabai.rb [hunk 3]
```diff
@@ -39,41 +39,15 @@ def caveats; <<~EOS
...
-      <string>#{var}/log/yabai/yabai.err.log</string>
-      <key>ProcessType</key>
-      <string>Interactive</string>
-      <key>Nice</key>
-      <integer>-20</integer>
-    </dict>
-    </plist>
-    EOS
+  service do
+    run opt_bin/"yabai"
```
Comment (mvllow) [inline yabai.rb:43]: This could probably be ignored, just suggesting for consistency.

```suggestion
    run "#{opt_bin}/yabai"
```

**Review by mvllow — CHANGES_REQUESTED** [review]
A lot of this was discovered through trial and error so some suggestions may be overkill or unnecessary but this does work when installing locally via `brew install ./yabai.rb`. Hope this helps!

File: yabai.rb [hunk 4]
```diff
@@ -39,41 +39,15 @@ def caveats; <<~EOS
...
+    require_root true
```
Comment (mvllow) [inline yabai.rb:44]: Does this need root?

```suggestion
```

File: yabai.rb [hunk 5]
```diff
@@ -39,41 +39,15 @@ def caveats; <<~EOS
...
+    environment_variables HOMEBREW_PREFIX/"bin:/usr/bin:/bin:/usr/sbin:/sbin"
+    keep_alive true
+    interval 30
```
Comment (mvllow) [inline yabai.rb:47]: > seem like this only request when run_type is set to interval

Suggestion per @hunterliao29 on #27 

```suggestion
```

File: yabai.rb [hunk 6]
```diff
@@ -39,41 +39,15 @@ def caveats; <<~EOS
...
+    log_path var/"log/yabai/yabai.out.log"
```
[the remark this answers is not in the document]
Comment (CKolkey) [reply yabai.rb:48]: Ruby would parse this as `var / "...string content"`, so the division operator. Concatenation would be `var + "/....", but interpolating it like you did is ideal.

**Comment by koekeishiya** [comment]
Track #31 instead.

**Pull request closed without merging**